prqlc/prqlc/tests/integration/dbs/protocol.rs

```diff
@@ -2,89 +2,50 @@ use anyhow::Result;
 use connector_arrow::api::{ResultReader, Statement};
 use connector_arrow::arrow;
 use connector_arrow::arrow::record_batch::RecordBatch;
-use serde::{Deserialize, Serialize};
 
-#[derive(Serialize, Deserialize)]
-#[serde(rename_all = "lowercase", tag = "type", content = "params")]
-pub enum DbProtocol {
-    DuckDb,
-    MsSql,
-    MySql { url: String },
-    Postgres { url: String },
-    SQLite,
+fn read_to_batch<'a>(reader: impl ResultReader<'a>) -> Result<RecordBatch> {
+    let batches = reader.into_iter().collect::<Result<Vec<_>, _>>()?;
+    let schema = batches.first().unwrap().schema_ref();
+    Ok(arrow::compute::concat_batches(schema, &batches)?)
 }
 
-pub trait DbProtocolHandler: Send {
+pub(crate) trait DbProtocol: Send {
     fn query(&mut self, sql: &str) -> Result<RecordBatch>;
-
     fn execute(&mut self, sql: &str) -> Result<()>;
 }
 
-#[cfg(feature = "test-dbs")]
-pub mod sqlite {
-    use super::*;
-
-    #[allow(dead_code)]
-    pub fn init() -> Box<dyn DbProtocolHandler> {
-        let conn = rusqlite::Connection::open_in_memory().unwrap();
-        let conn_ar = connector_arrow::sqlite::SQLiteConnection::new(conn);
-
-        Box::new(conn_ar)
+impl DbProtocol for connector_arrow::sqlite::SQLiteConnection {
+    fn query(&mut self, sql: &str) -> Result<RecordBatch> {
+        let mut statement = connector_arrow::api::Connector::query(self, sql)?;
+        let reader = statement.start([])?;
+        read_to_batch(reader)
     }
 
-    impl DbProtocolHandler for connector_arrow::sqlite::SQLiteConnection {
-        fn query(&mut self, sql: &str) -> Result<RecordBatch> {
-            let mut statement = connector_arrow::api::Connector::query(self, sql)?;
-            let reader = statement.start([])?;
-            read_to_batch(reader)
-        }
-
-        fn execute(&mut self, sql: &str) -> Result<()> {
-            self.inner_mut().execute(sql, ())?;
-            Ok(())
-        }
+    fn execute(&mut self, sql: &str) -> Result<()> {
+        self.inner_mut().execute(sql, ())?;
+        Ok(())
     }
 }
 
-#[cfg(feature = "test-dbs")]
-pub mod duckdb {
```

Comment:
FWIW @aljazerzen I like this pattern of generic modules ... in OCaml! I don't think it's rust-ic — if we're doing the same methods on each we can use a trait?